extension/entitystore/extension.go

```diff
@@ -94,9 +94,13 @@ func (e *EntityStore) Start(ctx context.Context, host component.Host) error {
 	e.serviceprovider = newServiceProvider(e.mode, e.config.Region, &e.ec2Info, e.metadataprovider, getEC2Provider, ec2CredentialConfig, e.done, e.logger)
 	switch e.mode {
 	case config.ModeEC2:
-		e.ec2Info = *newEC2Info(e.metadataprovider, e.done, e.config.Region, e.logger)
+		e.ec2Info = *newEC2Info(e.metadataprovider, e.kubernetesMode, e.done, e.config.Region, e.logger)
 		go e.ec2Info.initEc2Info()
-		go e.serviceprovider.startServiceProvider()
+		// Instance metadata tags is not usable for EKS nodes
+		// https://github.com/kubernetes/cloud-provider-aws/issues/762
+		if e.kubernetesMode == "" {
+			go e.serviceprovider.startServiceProvider()
```

Comment:
Does `e.serviceprovider.startServiceProvider()` do anything beneficial on EKS? Seems strange that we would have to do this just to avoid the IMDS check

Reply:
I don't think so. `startServiceProvider` scrapes IAM role or service name from instance tags so they can be used for service names. In Kubernetes environment neither of these will be used because they have lower priority than `Instrumentation` or `K8sWorkload`. 

Code wise, I only see serviceprovider public function called in `awsentity` processor if we are strictly on EC2: https://github.com/aws/amazon-cloudwatch-agent/blob/161cd5ba722e09ab0aa3b0a6d3f40e3a06e6aa1d/plugins/processors/awsentity/processor.go#L214